File: evidence/generated-systems/angular/crm-on-angular/src/app/models/Contract.js

// Define collection and schema for Contract
export interface Contract {
    contractNumber:
	type : string
    startDate:
	type : Date
    endDate:
	type : Date
    renewalTermMonths:
	type : number
    autoRenew:
	type : boolean
    Organization:
	type : Schema.Types.ObjectId
    Account:
	type : Schema.Types.ObjectId
    Owner:
	type : Schema.Types.ObjectId
    Orders:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Order' }]
    Cases:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Case_' }]
    Status:
 	type : String
#
    collection: 'contracts'
}

File: evidence/generated-systems/rubyonrails/crm-on-rails/crmonrails/app/models/contract.rb
class Contract < ApplicationRecord
  enum Status: [:Draft, :Active, :Suspended, :Expired, :Terminated, :Renewed]


  has_many :Organization, class_name: 'Organization'
  has_many :Account, class_name: 'Account'
  has_many :Owner, class_name: 'User'
  has_many :Orders, class_name: 'Order'
  has_many :Cases, class_name: 'Case_'

end
